cmd/clusterctl/pkg/log/log.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Package log mirrors the controller runtime approach to logging, by defining a global logger
+that defaults to NullLogger.
+
+User can SetCustom logger by calling log.SetLogger.
+*/
+package log
+
+import (
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// SetLogger sets a concrete logging implementation for all deferred Loggers.
+func SetLogger(l logr.Logger) {
+	Log.Fulfill(l)
+}
+
+// Log is the base logger used by kubebuilder.  It delegates
+// to another logr.Logger.  You *must* call SetLogger to
+// get any actual logging.
+var Log = log.NewDelegatingLogger(log.NullLogger{})
```

Comment:
Do we need the DelegatingLogger?

Reply:
I like the fact that the DelegatingLogger is explicitly providing a default null logger that works well with unit tests/as a default behavior from the library and it can also host user-provided loggers.

This can be achieved in other ways, but since we are already importing controller-runtime, I don't see backsides on taking benefit of this component as well

Reply:
sgtm! thanks for your thoughts